### ISSUE TITLE
zrangebylex and zadd are not encoding values the same way

### ISSUE THREAD

**Issue by madgnome** [mp911de/lettuce#361]
`zadd` encodes value using `addValue` but `zrangebylex` and co. encodes them as string using `add(String s)`. As a result it's impossible to get results for unicode values out of the ascii range.

```
    // Add this test to SortedSetCommandTest
    @Test
    public void zrangebylexUnicode() throws Exception {

        redis.zadd(key, 0d, "a");
        redis.zadd(key, 0d, "富");

        List<String> result = redis.zrangebylex(key, "[a", "[a\uFFFF");
        assertThat(result).hasSize(1);

        result = redis.zrangebylex(key, "[富", "[富\uFFFF");
        assertThat(result).hasSize(1); // This will fail
    }
```

**Comment by mp911de** [comment]
Thanks for the bug report. As workaround, you please use the custom command API:

```
redis.zadd(key, 0d, "a");
redis.zadd(key, 0d, "富");

redis.zrangebylex(key, "[富", "[富\uFFFF");

List<String> values = redis.dispatch(ZRANGEBYLEX, new ValueListOutput<>(utf8StringCodec),
        new CommandArgs<>(utf8StringCodec)
                .addKey(key)
                .add("[富".getBytes(StandardCharsets.UTF_8))
                .add("[富\uFFFF".getBytes(StandardCharsets.UTF_8)));
```

**Comment by mp911de** [comment]
I fixed this issue with #363 which required a new API.

**Comment by madgnome** [comment]
Awesome, thanks a lot!
